src/EventListener/InvalidVersionNumberListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\EventListener;
+
+use Libero\ApiProblemBundle\Event\CreateApiProblem;
+use Libero\ContentApiBundle\Exception\InvalidVersionNumber;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Translation\TranslatorInterface;
+use Throwable;
+
+final class InvalidVersionNumberListener
+{
+    use TranslatingApiProblemListener;
+
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    protected function supports(Throwable $exception) : bool
+    {
+        return $exception instanceof InvalidVersionNumber;
+    }
+
+    protected function createStatus(CreateApiProblem $event) : int
+    {
+        return Response::HTTP_BAD_REQUEST;
+    }
+
+    protected function getTitleKey(CreateApiProblem $event) : string
+    {
+        return 'libero.content.item.invalid_version.title';
+    }
+
+    protected function getDetailsKey(CreateApiProblem $event) : ?string
+    {
+        return 'libero.content.item.invalid_version.details';
+    }
```

Comment:
Keen to reduce the boilerplate here:

One option is the convention-over-configuration way, which calculates values from things such as the class name, and makes discovery difficult.

The other could be to coalesce the various methods into a single one that returns a Value Object:
```
    protected function getDetails(CreateApiProblem $event) : TranslationDetails
    {
        return new TranslationDetails([
            'status' => ...,
            'title' => 'libero.content.item.invalid_version.title',
            'details' => 'libero.content.item.invalid_version.details',
            ...
        ]);
    }
```

Reply:
Yeah, looks like a good approach.